tests/test_validation_framework/test_metric_calculators_adapters.py

```diff
@@ -0,0 +1,142 @@
+import unittest
+
+import pandas as pd
+import numpy as np
+
+from pytesmo.validation_framework.metric_calculators_adapters import (
+    GenericDatetime, TsDistributor)
+from datetime import datetime
+
+
+class Test_GenericDateTime(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.now = datetime.now()
+        self.past = datetime(1900, 1, 2, 3, 4, 5)
+        self.future = datetime(2104, 6, 7, 8, 9, 10)
+        self.gnow = GenericDatetime(self.now.month, self.now.day,
+                                    self.now.hour, self.now.minute,
+                                    self.now.second)
+
+    def test_comparisons(self):
+        assert self.gnow > GenericDatetime.from_datetime(self.past)
+        assert self.gnow < GenericDatetime.from_datetime(self.future)
+        assert self.gnow == self.gnow
+        assert self.gnow <= self.gnow
+        assert self.gnow >= self.gnow
+
+    def test_doy(self):
+        assert GenericDatetime.from_datetime(self.future).doy == 159
+        try:
+            _ = datetime(self.now.year, 2, 29)
+            assert self.now.timetuple().tm_yday == self.gnow.doy
+        except ValueError:
+            if self.now > datetime(self.now.year, 2, 28):
+                assert self.now.timetuple().tm_yday + 1 == self.gnow.doy
+            else:
+                assert self.now.timetuple().tm_yday == self.gnow.doy
```

Comment:
In general I think it's better if we set a few dates to test instead of using `datetime.now`, so that the actual test code that is run is the same all the time.

Could you also add a comment about why the try except statement is necessary, and for the if-else statement?

Reply:
agree, will update it